### PR TITLE
feat(extras): Add extra for angular lspconfig

### PR DESCRIPTION
**Pull request by radvil**
Since treesitter has better support for angular parser recently that works for the legacy and the latest version of it, I am happy to suggest this extra configuration that works for well for me without any issue. And since lack of references about how to config the angular lsp inside of neovim out there, this will work fine as a starting point.

### CODE REVIEW

**Comment by radvil** [comment]
Hi @folke I think the new `vtsls` lsp config pairs really well with this configuration, I am experiencing it's running smooth now. Can you review this ?

File: lua/lazyvim/plugins/extras/lang/angular.lua
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/angular.lua:4]: Can you remove the filetypes here?
Otherwise angular will show up as recommended for any typescript/html file.
Having the root dir detection is enough

File: lua/lazyvim/plugins/extras/lang/angular.lua
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/angular.lua:54]: You shoyld probably remove that warn=false?
I've only added that because vue/astro need an upstream mason PR merged to make it work

**Review by folke — CHANGES_REQUESTED** [review]

File: lua/lazyvim/plugins/extras/lang/angular.lua
Comment (radvil) [reply lua/lazyvim/plugins/extras/lang/angular.lua:4]: nice, thanks for the response. I ll update the PR, maybe I ll also remove the `nx.json` ? because nx can also be used to other frameworks.

File: lua/lazyvim/plugins/extras/lang/angular.lua
Comment (radvil) [reply lua/lazyvim/plugins/extras/lang/angular.lua:54]: Thanks for this info, I ll update my loval config first and trynna play around with my angular codebases about one hour, then I ll update the PR again. Thanks..

**Comment by radvil** [comment]
Hi @folke, just wanna let you know I've updated this PR based your recommendation. Also with a slight different on 
```typescript
enableForWorkspaceTypeScriptVersions = false
```
as per recommendation on the README here https://github.com/angular/vscode-ng-language-service to opt typescript version bundled with the language server itself as the default. However I want to try this config for on daily basis, maybe you should not merge this now, I ll report tomorrow if I found anything doesn't work as expected, but for now I am not having any issue with it.

**Comment by folke** [comment]
Looks good. Thanks!

**Comment by radvil** [comment]
> !

Thank, I love ur work man..

**Comment by rofrol** [comment]
@radvil I have enabled the plugin. Restarted nvim.

But it seems it does not work for me. I am looking at these features https://angular.dev/tools/language-service

- [Autocompletion](https://angular.dev/tools/language-service#autocompletion) - works without this plugin
- [Error checking](https://angular.dev/tools/language-service#error-checking) - does not work, no error shown
- [Quick info and navigation](https://angular.dev/tools/language-service#quick-info-and-navigation) - no info popup, `<leader>gd` does nothing.

**Comment by radvil** [comment]
> @radvil I have enabled the plugin. Restarted nvim.
> 
> But it seems it does not work for me. I am looking at these features https://angular.dev/tools/language-service
> 
> * [Autocompletion](https://angular.dev/tools/language-service#autocompletion) - works without this plugin
> * [Error checking](https://angular.dev/tools/language-service#error-checking) - does not work, no error shown
> * [Quick info and navigation](https://angular.dev/tools/language-service#quick-info-and-navigation) - no info popup, `<leader>gd` does nothing.
> 
> EDIT: Maybe I need to enable strictTemplates in tsconfig.json?

Thanks for your feedback on this, finally another angular developer ..😁

Strangely enough, because mine works just fine even without "strictTemplate".  Which version of angular ur project currently use?


https://github.com/LazyVim/LazyVim/assets/36059968/35d3277d-e051-4bbb-988b-60d4196c6508

**Comment by radvil** [comment]
I am not so sure if I did this correctly, maybe we should add description for this as experimental? what do you think @folke ? I am not really a wizard in this, but I try to do it best. Thanks again for your feedback @rofrol.

**Pull request merged**
